full-backend-tests/src/test/java/org/graylog/datanode/ClientCertResourceIT.java

```diff
@@ -0,0 +1,162 @@
+import org.graylog.security.certutil.CertConstants;
+import org.graylog.security.certutil.csr.InMemoryKeystoreInformation;
+import org.graylog.security.certutil.csr.KeystoreInformation;
+import org.graylog.testing.completebackend.apis.GraylogApiResponse;
+import org.graylog.testing.completebackend.apis.GraylogApis;
+import org.graylog.testing.containermatrix.SearchServer;
+import org.graylog.testing.containermatrix.annotations.ContainerMatrixTest;
+import org.graylog.testing.containermatrix.annotations.ContainerMatrixTestsConfiguration;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+
+@ContainerMatrixTestsConfiguration(searchVersions = SearchServer.DATANODE_DEV,
+                                   additionalConfigurationParameters = {
+                                           @ContainerMatrixTestsConfiguration.ConfigurationParameter(key = "GRAYLOG_DATANODE_INSECURE_STARTUP", value = "false"),
+                                           @ContainerMatrixTestsConfiguration.ConfigurationParameter(key = "GRAYLOG_SELFSIGNED_STARTUP", value = "true"),
+                                           @ContainerMatrixTestsConfiguration.ConfigurationParameter(key = "GRAYLOG_ELASTICSEARCH_HOSTS", value = ""),
+                                   })
+public class ClientCertResourceIT {
+
+    private final GraylogApis api;
+
+    public ClientCertResourceIT(GraylogApis api) {
+        this.api = api;
+    }
+
+    @ContainerMatrixTest
+    void generateClientCert() throws Exception {
+        // these roles are supported: all_access,security_rest_api_access,readall
+        final ValidatableResponse clientCertResponse = api.post("/ca/clientcert", """
+                {
+                    "principal": "admin",
+                    "role": "all_access",
+                    "password": "asdfgh",
+                    "certificate_lifetime": "P6M"
+                }
+                """, Response.Status.OK.getStatusCode());
+
+        final GraylogApiResponse parsedResponse = new GraylogApiResponse(clientCertResponse);
+        final X509Certificate caCertificate = decodeCert(parsedResponse.properJSONPath().read("ca_certificate"));
+        final PrivateKey privateKey = decodePrivateKey(parsedResponse.properJSONPath().read("private_key"));
+        final X509Certificate certificate = decodeCert(parsedResponse.properJSONPath().read("certificate"));
+
```

Comment:
How about adding some additional tests here for the certificate details (since there is currently no test for `ClientCertGenerator`)?
```suggestion

        assertThat(certificate.getIssuerX500Principal().getName()).isEqualTo("CN=Graylog CA");
        assertThat(certificate.getSubjectX500Principal().getName()).isEqualTo("CN=admin");
        LocalDate expires = certificate.getNotAfter()
                .toInstant()
                .atZone(ZoneId.systemDefault())
                .toLocalDate();
        assertThat(expires).isBefore(LocalDate.now().plusMonths(6).plusDays(2));
        assertThat(expires).isAfter(LocalDate.now().plusMonths(6).minusDays(2));
```